CodeGen/src/OptimizeConstProp.cpp

```diff
@@ -726,6 +727,9 @@ static void constPropInInst(ConstPropState& state, IrBuilder& build, IrFunction&
                             arg->cmd == IrCmd::UNM_VEC)
                             tag = LUA_TVECTOR;
                     }
+
+                    if (FFlag::LuauCodegenLoadTVTag && arg->cmd == IrCmd::LOAD_TVALUE && arg->c.kind != IrOpKind::None)
+                        tag = function.tagOp(arg->c);
```

Comment:
This hint should not be associated strictly with STORE_TVALUE.
LOAD_TVALUE handler should update the tag in the state.

Reply:
:sweat_smile: fwiw we use undef/tag in another place, but LOAD_TVALUE already used none/int. I am also assuming "none" is exactly equivalent to "0", which is why I specify 0 instead of "none" so that we don't have gaps in IR op arguments.

Reply:
LOAD_TVALUE loads from a constant to IR value though, and tags in the state are associated with registers -- so LOAD_TVALUE by itself can't record this unless I'm missing something.

Reply:
Yeah, you're right, it's a limitation.
I didn't like my block for vector above, so I didn't expect that more stuff is going to be added.

Usually stuff like NEW_TABLE is followed by STORE_TAG to help out, but vectors are a special case since they are operated on like on a single value.